crates/egui-winit/src/window_settings.rs

```diff
@@ -86,4 +83,58 @@ impl WindowSettings {
             *size = size.at_most(max_size);
         }
     }
+
+    pub fn clamp_window_to_sane_position<E>(
+        &mut self,
+        event_loop: &winit::event_loop::EventLoopWindowTarget<E>,
+    ) {
+        use std::ops::{Add, AddAssign, MulAssign, Sub};
+
+        if let (Some(position), Some(inner_size)) = (&mut self.position, &self.inner_size_points) {
+            let monitors = event_loop.available_monitors();
+            // default to primary monitor, in case the correct monitor was disconnected.
+            let mut monitor: winit::monitor::MonitorHandle = match event_loop.primary_monitor() {
+                Some(m) => m,
+                // No primary monitor, grab index 0 instead
+                _ => {
+                    match event_loop.available_monitors().nth(0) {
+                        Some(m) => m,
+                        None => return, // No monitors?
+                    }
+                }
+            };
+            for _monitor in monitors {
+                let monitor_x_range =
+                    _monitor.position().x..(_monitor.position().x + _monitor.size().width as i32);
+                let monitor_y_range =
+                    _monitor.position().y..(_monitor.position().y + _monitor.size().height as i32);
+
+                if monitor_x_range.contains(&(position.x as i32))
+                    && monitor_y_range.contains(&(position.y as i32))
+                {
+                    monitor = _monitor
+                }
+            }
+
+            let mut inner_size = inner_size.clone(); // Shadow inner_size, we don't want to modify the original
+            inner_size.mul_assign(monitor.scale_factor() as f32);
+            // Add size of title bar. This is 32 px by default in Win 10/11.
+            #[cfg(windows)]
+            inner_size.add_assign(egui::Vec2::new(0.0, 32.0 * monitor.scale_factor() as f32));
+            let monitor_position =
+                egui::Pos2::new(monitor.position().x as f32, monitor.position().y as f32);
+            let monitor_logical_size = monitor.size();
+            *position = position.clamp(
+                monitor_position,
+                // To get the maximum position, we get the rightmost corner of the display, then subtract
+                // the size of the window to get the bottom right most value window.position can have.
+                monitor_position
+                    .add(egui::Vec2::new(
+                        monitor_logical_size.width as f32,
+                        monitor_logical_size.height as f32,
+                    ))
+                    .sub(inner_size),
```

Comment:
```suggestion
                monitor_position + monitor_logical_size - inner_size_points,
```

note that what winit call "logical_size" is what egui call "points".

Reply:
Cannot directly add (what was) monitor_logical_size to monitor_position because Pos2 does not impl Add for PhysicalSize<T>, the manual cast is necessary.

Reply:
`monitor_position + egui::Vec2::new(
                    monitor_size.width as f32,
                    monitor_size.height as f32,
                ) - inner_size_pixels,`